### PR TITLE
Internal Transfer feature.

### PR DESCRIPTION
**Pull request by hbulgarini**
Related with #213.

This PR includes the finished v1 of the internal Transfer feature. 

Also i have renamed in the code and in the app the term `Local` by `Internal` which is more aligned with the feature description on the issue #213.

### CODE REVIEW

**Comment by hbulgarini** [comment]
@tomusdrw @goldsteinsveta : I have done a minor change on the design regarding the internal receiver field. I added a companion field below the input similar as what we have now. I saw the need of adding this addiotional field below the input because when the user enters a generic account format ( 42 ss58Prefix ) i think is useful for the user to see the receiver account formatted  with the corresponding chain prefix:

Providing native account:
Screen Shot 2021-08-10 at 11 34 39

Providing generic account:
Screen Shot 2021-08-10 at 11 34 53

**Review by wirednkod — CHANGES_REQUESTED** [review]

File: src/components/Sender.tsx [hunk 1]
```diff
@@ -71,6 +71,9 @@ const useStyles = makeStyles((theme) => ({
   },
   accountCompanion: {
     ...styleAccountCompanion(theme)
+  },
+  sender: {
+    minHeight: '100px'
```
[the remark this answers is not in the document]
Comment (hbulgarini) [reply src/components/Sender.tsx:76]: ok! something like `theme.spacing(10)` ?

File: src/contexts/GUIContextProvider.tsx [hunk 2]
```diff
@@ -65,6 +66,8 @@ export function GUIContextProvider({ children }: GUIContextProviderProps): React
 
   const value = { drawer, setDrawer, actions, action, setAction, isBridged, setBridged };
 
+  useEffect(() => setBridged(true), [action]);
```
[the remark this answers is not in the document]
Comment (hbulgarini) [reply src/contexts/GUIContextProvider.tsx:69]: i'm not sure i'm following your comment. When the action variable is changed, this useEffect is triggered reseting the bridged value to true every time.

File: src/components/BridgedLocalWrapper.tsx [hunk 3]
```diff
@@ -20,16 +20,21 @@ import { useGUIContext } from '../contexts/GUIContextProvider';
...
+  }
+
   if (!isBridged) {
     if (blurred) return <div style={{ opacity: '20%' }}>{children}</div>;
   }
 
-  if (isBridged) {
+  if (isBridged && !showInternal) {
     return children;
   }
```
[the remark this answers is not in the document]
Comment (hbulgarini) [reply src/components/BridgedLocalWrapper.tsx:39]: cool ... last if misses the isBridged:

`  if (isBridged && !showInternal) return children;`

File: src/contexts/GUIContextProvider.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (hbulgarini) [reply src/contexts/GUIContextProvider.tsx:69]: ahhh maybe as the code is inline you cannot read the `[action]` dependency?

```
useEffect(() => {
         setBridged(true)
      }
   ,[action]);
```

File: src/components/Sender.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (wirednkod) [reply src/components/Sender.tsx]: Exactly

File: src/contexts/GUIContextProvider.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (wirednkod) [reply src/contexts/GUIContextProvider.tsx:69]: No I saw the dependency. It just looks weird that the dependent value is not used inside the useEffect.
Afaiu though you want the setBridge to run only when action changes, right?

**Review by wirednkod — APPROVED** [review]

**Pull request merged**